src/xdr/Stellar-ledger-entries.x

```diff
@@ -28,6 +28,17 @@ enum AssetType
     ASSET_TYPE_CREDIT_ALPHANUM12 = 2
 };
 
+union AssetCode switch (AssetType type)
```

Comment:
The issue with `NonNativeAsset` is that it sounds like it includes the issuer, since that information is included in `Asset`, and `NonNativeAssetCode` is redundant because asset codes only apply to non-native assets. @leighmcculloch and I thought `AssetCode` made the most sense here.

Reply:
+1 We debated the name of this field for a while, and AssetCode seems like the simplest to understand without implying things that are false. Ideally it is given a name that communicates the intent that it be used whenever you need to provide the fields for a non-native asset not including the issuer.

Reply:
Oh I see - I guess that boat sailed a long time ago as `AssetCode` is just derived from `AssetCode4/12` so it's probably fine and just reinforcing an existing concept